src/libs/imageManipulator/index.js

```diff
@@ -0,0 +1,144 @@
+ * @param {Number} degrees
+ * @returns {Object}
+ */
+function rotateCanvas(canvas, degrees) {
+    const {width, height} = sizeFromAngle(canvas.width, canvas.height, degrees);
+
+    const result = document.createElement('canvas');
+    result.width = width;
+    result.height = height;
+
+    const context = result.getContext('2d');
+    context.translate(result.width / 2, result.height / 2);
+
+    const radians = (degrees * Math.PI) / 180;
+    context.rotate(radians);
+
+    context.drawImage(canvas, -canvas.width / 2, -canvas.height / 2, canvas.width, canvas.height);
+
+    return result;
+}
+
+/**
+ * Creates new cropped canvas and returns it
+ *
+ * @param {Object} canvas
+ * @param {Object} options
+ * @returns {Object}
+ */
+function cropCanvas(canvas, options) {
+    let {
+        originX = 0, originY = 0, width = 0, height = 0,
+    } = options;
+    const clamp = (value, max) => Math.max(0, Math.min(max, value));
+
+    width = clamp(width, canvas.width);
+    height = clamp(height, canvas.height);
+    originX = clamp(originX, canvas.width);
+    originY = clamp(originY, canvas.height);
+
+    width = Math.min(originX + width, canvas.width) - originX;
+    height = Math.min(originY + height, canvas.height) - originY;
+
+    const result = document.createElement('canvas');
+    result.width = width;
+    result.height = height;
+
+    const context = result.getContext('2d');
+    context.drawImage(canvas, originX, originY, width, height, 0, 0, width, height);
+
+    return result;
+}
+
+/**
+ * @param {Object} canvas
+ * @param {Object} options
+ * @returns {Promise<File>}
+ */
+function convertCanvasToFile(canvas, options = {}) {
+    return new Promise((resolve) => {
+        canvas.toBlob((blob) => {
+            const file = new File([blob], `${options.name || 'fileName'}.jpg`, {type: 'image/jpeg'});
+            file.uri = URL.createObjectURL(file);
+            resolve(file);
+        });
+    });
+}
+
+/**
+ * Loads image from specified url
+ *
+ * @param {String} uri
+ * @returns {Promise<Object>}
+ */
+function loadImageAsync(uri) {
+    return new Promise((resolve, reject) => {
+        const imageSource = new Image();
+        imageSource.crossOrigin = 'anonymous';
+        const canvas = document.createElement('canvas');
+        imageSource.onload = () => {
+            canvas.width = imageSource.naturalWidth;
+            canvas.height = imageSource.naturalHeight;
+
+            const context = canvas.getContext('2d');
+            context.drawImage(imageSource, 0, 0, imageSource.naturalWidth, imageSource.naturalHeight);
+
+            resolve(canvas);
+        };
+        imageSource.onerror = () => reject(canvas);
+        imageSource.src = uri;
+    });
+}
+
+/**
+ * Crops and rotates the image on web
+ *
+ * @param {String} uri
+ * @param {Object} actions
+ * @param {Object} options
+ * @returns {Promise<Object>} Returns cropped and rotated image
+ */
+function imageManipulator(uri, actions, options) {
```

Comment:
The name of this method is a little off, and would suggest more like it's returning some kind of interface. Instead, consider renaming it to `cropOrRotateImage()`